src/optimizer.rs

```diff
@@ -48,6 +48,18 @@ pub fn optimize(program: &[Instruction]) -> Result<Vec<Instruction>> {
         Move(total)
       }
 
+      JumpIfZero(loop_end) => {
+        match program.get(index + 1..=loop_end) {
```

Comment:
Let's match `program[index + 1..]` with `[Add(-1), JumpIfNonZero(_)]`. This will make optimizer more extensible in the future.

Reply:
Incorrect suggestion. Subslice patterns are not yet stabilized, so this requires nightly.